src/testsets.jl

```diff
@@ -28,56 +24,81 @@ function finish(ts::ConcurrentTestSet)
     if Test.get_testset_depth() > 0
         # Attach this test set to the parent test set
         parent_ts = Test.get_testset()
-        record(parent_ts, ts.dts)
+        record(parent_ts, ts)
         return ts
     end
     finish(ts.dts)
-    return ts.dts
+    return ts
 end
 
 function add_test_ref(testset::ConcurrentTestSet, test_ref)
     return push!(testset.tests, test_ref)
 end
 
-# Handle attempted use outside of a ConcurrentTestSEt
+# Handle attempted use outside of a ConcurrentTestSet
 function add_test_ref(testset::AbstractTestSet, test_ref)
     return fetch(test_ref)
 end
 
-# Wrap a DefaultTestSet. Results are recorded, but not printed if nested=true.
+# Wrap a DefaultTestSet with some behavior specific to @test_rel.
+# 
+# Results are recorded, but not printed if nested=true.
 # This is helpful when used in a ConcurrentTestSet where the parent
-# linkage is lost due to the concurrency
+# linkage is lost due to the concurrency.
+#
+# Additionally the whole test set can be checked for broken-ness, this is
+# needed as a test rel desugars into multiple checks and we don't know which
+# is expected not to work.
 mutable struct TestRelTestSet <: AbstractTestSet
     dts::Test.DefaultTestSet
     nested::Bool
+    broken::Bool
+    broken_found::Bool
 
-    TestRelTestSet(desc; nested=false) = new(Test.DefaultTestSet(desc), nested)
+    TestRelTestSet(desc; nested=false, broken=false) = 
+        new(Test.DefaultTestSet(desc), nested, broken, false)
 end
 
 record(ts::TestRelTestSet, child::AbstractTestSet) = record(ts.dts, child)
+record(ts::Test.DefaultTestSet, child::TestRelTestSet) = record(ts, child.dts)
 record(ts::TestRelTestSet, res::Test.Result) = record(ts.dts, res)
 
-# log these - by default they get printed to stdout
+# Flip to broken if expected, if not, log them (recording to dts goes to stdout)
```

Comment:
If the other comments weren't so good, I wouldn't have noticed, but this could be clarified a bit. Something like
```suggestion
# Record result without sending to stdout. If broken results are expected, flip Fail/Error to Broken then record.
```

Reply:
that's not quite right either - tried again